### ISSUE TITLE
Object Gio.Settings has been already deallocated - impossible to access to it

### ISSUE THREAD

**Issue by bittner** [paradoxxxzero/gnome-shell-system-monitor-applet#449]
GNOME Shell is writing the following error log entries to `/var/log/syslog` every 3 seconds. Is that something we can fix in this extension?

I'm using Ubuntu 18.04 LTS (development version) with GNOME 3.28.1, GNOME Shell 3.28.0, and the current version of this extension available from [extensions.gnome.org](https://extensions.gnome.org/extension/120/system-monitor/).

```
gnome-shell[3186]: Object Gio.Settings (0x55e3cf9f4eb0), has been already deallocated - impossible to access to it. This might be caused by the fact that the object has been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs
gnome-shell[3186]: message repeated 3 times: [ Object Gio.Settings (0x55e3cf9f4eb0), has been already deallocated - impossible to access to it. This might be caused by the fact that the object has been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs]
org.gnome.Shell.desktop[3186]: == Stack trace for context 0x55e3ce0f40c0 ==
org.gnome.Shell.desktop[3186]: #0 0x7fffa3cbc670 b   /home/peter/.local/share/gnome-shell/extensions/system-monitor@paradoxxx.zero.gmail.com/extension.js:2009 (0x7f3eff1c12b8 @ 39)
org.gnome.Shell.desktop[3186]: #1 0x7fffa3cbc6d0 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7f3f4c1b5de0 @ 71)
org.gnome.Shell.desktop[3186]: #2 0x7fffa3cbc6d0 I   /home/peter/.local/share/gnome-shell/extensions/system-monitor@paradoxxx.zero.gmail.com/extension.js:928 (0x7f3eff1bd098 @ 70)
org.gnome.Shell.desktop[3186]: #3 0x7fffa3cbc730 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7f3f4c1b5de0 @ 71)
org.gnome.Shell.desktop[3186]: #4 0x7fffa3cbc7d0 I   self-hosted:913 (0x7f3f4c1ebf78 @ 346)
```

The error message references lines [2009](https://github.com/paradoxxxzero/gnome-shell-system-monitor-applet/blob/master/system-monitor%40paradoxxx.zero.gmail.com/extension.js#L2009) and [928](https://github.com/paradoxxxzero/gnome-shell-system-monitor-applet/blob/master/system-monitor%40paradoxxx.zero.gmail.com/extension.js#L928) in `extension.js`.

**Comment by franglais125** [comment]
This is something that was already fixed in git master [0c95c03f0d434b9ce0b18b9a681a41fd8d366ac2].

@j0rd what do you mean by `master`? Are you sure you are running the latest git code?

**Comment by bittner** [comment]
> This is something that was already fixed in git master [0c95c03].

How long will it take to get the changes published, so that they are installed also vio extension.gnome.org?

**Comment by piegamesde** [comment]
See #443. (Spoiler: probably a long time)

**Comment by j0rd** [comment]
@franglais125 Ubuntu 18.04 LTS master (sorry, I mean I'm not running Ubuntu -beta)

Confusing and stupid for me to add that, sorry for the confusion.

PS. I had to disable this extension as it ate up like consistent 5-10% CPU usage, even after I patched this problem myself manually.

**Comment by gsalgado** [comment]
Any chance we can get a new version pushed for review to extensions.g.o with this fix? I don't see system-monitor in https://extensions.gnome.org/review/ so I'm guessing one hasn't been pushed yet?

**Comment by jankatins** [comment]
I have the same, but only after I locked my computer.

**Comment by ciofeca** [comment]
Same problem here, on Ubuntu 18.04 and its stock Gnome (3.28.1). I found out when noticed _/var/log/journal_ here grew to 5 gigabytes (!!!). That error (at line 2009 and 928) happens even if using the last committed _extension.js_.

**Comment by nloyola** [comment]
Are you sure @jankatins? I just installed from master and the problem is not fixed.

**Comment by zombeer** [comment]
Confirming the prob.
Bionic Beaver 18.04, GNOME Shell 3.28.1

**Comment by VadimKulagin** [comment]
I have a similar problem.
But errors are written to syslog not at intervals, but when I point the cursor at the indicators:
```
Jul  2 15:22:47 computer gnome-shell[2982]: Object Gio.Settings (0x55f9c796a520), has been already deallocated - impossible to access to it. This might be caused by the fact that the object has been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs
Jul  2 15:22:47 computer org.gnome.Shell.desktop[2982]: == Stack trace for context 0x55f9c5c7a340 ==
Jul  2 15:22:47 computer org.gnome.Shell.desktop[2982]: #0 0x55f9c6126678 i   /home/vadim/.local/share/gnome-shell/extensions/system-monitor@paradoxxx.zero.gmail.com/extension.js:752 (0x7ffad61d14d8 @ 29)
Jul  2 15:22:47 computer org.gnome.Shell.desktop[2982]: #1 0x55f9c61265d0 i   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:22:47 computer org.gnome.Shell.desktop[2982]: #2 0x55f9c6126510 i   self-hosted:917 (0x7ffb086f12b8 @ 394)
```

And as @zombeer said, when the lock screen is turned on, errors are written with insane speed.
```
Jul  2 15:27:00 computer gnome-shell[2982]: Object Gio.Settings (0x55f9c796a520), has been already deallocated - impossible to access to it. This might be caused by the fact that the object has been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: == Stack trace for context 0x55f9c5c7a340 ==
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #0 0x55f9c6126be0 i   /home/vadim/.local/share/gnome-shell/extensions/system-monitor@paradoxxx.zero.gmail.com/extension.js:2428 (0x7ffad61d6a28 @ 337)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #1 0x55f9c6126b20 i   resource:///org/gnome/shell/ui/extensionSystem.js:82 (0x7ffb0845a120 @ 431)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #2 0x55f9c6126aa0 i   resource:///org/gnome/shell/ui/extensionSystem.js:344 (0x7ffb0845abc0 @ 13)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #3 0x7ffe148f9830 b   self-hosted:251 (0x7ffb086c3ab0 @ 223)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #4 0x55f9c6126a20 i   resource:///org/gnome/shell/ui/extensionSystem.js:343 (0x7ffb0845ab38 @ 64)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #5 0x55f9c61269a0 i   resource:///org/gnome/shell/ui/extensionSystem.js:361 (0x7ffb0845ac48 @ 87)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #6 0x7ffe148fad30 b   resource:///org/gnome/gjs/modules/signals.js:128 (0x7ffb086d3230 @ 386)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #7 0x7ffe148fb540 b   resource:///org/gnome/shell/ui/sessionMode.js:205 (0x7ffb0806b670 @ 254)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #8 0x7ffe148fc930 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #9 0x55f9c6126860 i   resource:///org/gnome/shell/ui/sessionMode.js:167 (0x7ffb0806b450 @ 40)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #10 0x7ffe148fdc90 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #11 0x55f9c61267b8 i   resource:///org/gnome/shell/ui/screenShield.js:1280 (0x7ffb08050bc0 @ 188)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #12 0x7ffe148feff0 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #13 0x55f9c6126708 i   resource:///org/gnome/shell/ui/screenShield.js:1329 (0x7ffb08050c48 @ 391)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #14 0x7ffe14900350 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #15 0x55f9c6126680 i   resource:///org/gnome/shell/ui/shellDBus.js:462 (0x7ffb0805ad58 @ 67)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #16 0x7ffe149016b0 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #17 0x55f9c61265b8 i   resource:///org/gnome/gjs/modules/overrides/Gio.js:295 (0x7ffb086e5450 @ 614)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #18 0x55f9c6126510 i   resource:///org/gnome/gjs/modules/overrides/Gio.js:326 (0x7ffb086e5670 @ 34)
Jul  2 15:27:00 computer gnome-shell[2982]: System monitor applet disable
Jul  2 15:27:00 computer gnome-shell[2982]: Object .Gjs_AppIndicatorIconActor__1 (0x55f9c85e62f0), has been already finalized. Impossible to set any property to it.
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: == Stack trace for context 0x55f9c5c7a340 ==
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #0 0x7ffe148f5820 I   resource:///org/gnome/gjs/modules/_legacy.js:83 (0x7ffb086b5de0 @ 87)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #1 0x55f9c6126cc8 i   /usr/share/gnome-shell/extensions/ubuntu-appindicators@ubuntu.com/indicatorStatusIcon.js:93 (0x7ffad4155450 @ 58)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #2 0x7ffe148f6400 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #3 0x7ffe148f64c0 b   self-hosted:915 (0x7ffb086f12b8 @ 367)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #4 0x7ffe148f65b0 b   resource:///org/gnome/gjs/modules/signals.js:128 (0x7ffb086d3230 @ 386)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #5 0x55f9c6126c40 i   /usr/share/gnome-shell/extensions/ubuntu-appindicators@ubuntu.com/appIndicator.js:190 (0x7ffad4149120 @ 22)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #6 0x7ffe148f7200 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #7 0x55f9c6126b98 i   /usr/share/gnome-shell/extensions/ubuntu-appindicators@ubuntu.com/statusNotifierWatcher.js:219 (0x7ffad41456f8 @ 225)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #8 0x7ffe148f7de0 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #9 0x55f9c6126b20 i   /usr/share/gnome-shell/extensions/ubuntu-appindicators@ubuntu.com/extension.js:61 (0x7ffad413b9a0 @ 37)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #10 0x7ffe148f8ac0 b   resource:///org/gnome/shell/ui/extensionSystem.js:82 (0x7ffb0845a120 @ 431)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #11 0x55f9c6126aa0 i   resource:///org/gnome/shell/ui/extensionSystem.js:344 (0x7ffb0845abc0 @ 13)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #12 0x7ffe148f9830 b   self-hosted:251 (0x7ffb086c3ab0 @ 223)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #13 0x55f9c6126a20 i   resource:///org/gnome/shell/ui/extensionSystem.js:343 (0x7ffb0845ab38 @ 64)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #14 0x55f9c61269a0 i   resource:///org/gnome/shell/ui/extensionSystem.js:361 (0x7ffb0845ac48 @ 87)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #15 0x7ffe148fad30 b   resource:///org/gnome/gjs/modules/signals.js:128 (0x7ffb086d3230 @ 386)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #16 0x7ffe148fb540 b   resource:///org/gnome/shell/ui/sessionMode.js:205 (0x7ffb0806b670 @ 254)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #17 0x7ffe148fc930 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #18 0x55f9c6126860 i   resource:///org/gnome/shell/ui/sessionMode.js:167 (0x7ffb0806b450 @ 40)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #19 0x7ffe148fdc90 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #20 0x55f9c61267b8 i   resource:///org/gnome/shell/ui/screenShield.js:1280 (0x7ffb08050bc0 @ 188)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #21 0x7ffe148feff0 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #22 0x55f9c6126708 i   resource:///org/gnome/shell/ui/screenShield.js:1329 (0x7ffb08050c48 @ 391)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #23 0x7ffe14900350 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #24 0x55f9c6126680 i   resource:///org/gnome/shell/ui/shellDBus.js:462 (0x7ffb0805ad58 @ 67)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #25 0x7ffe149016b0 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #26 0x55f9c61265b8 i   resource:///org/gnome/gjs/modules/overrides/Gio.js:295 (0x7ffb086e5450 @ 614)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #27 0x55f9c6126510 i   resource:///org/gnome/gjs/modules/overrides/Gio.js:326 (0x7ffb086e5670 @ 34)
Jul  2 15:27:00 computer gnome-shell[2982]: Object .Gjs_AppIndicatorIconActor__1 (0x55f9cc259860), has been already finalized. Impossible to set any property to it.
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: == Stack trace for context 0x55f9c5c7a340 ==
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #0 0x7ffe148f5820 I   resource:///org/gnome/gjs/modules/_legacy.js:83 (0x7ffb086b5de0 @ 87)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #1 0x55f9c6126cc8 i   /usr/share/gnome-shell/extensions/ubuntu-appindicators@ubuntu.com/indicatorStatusIcon.js:93 (0x7ffad4155450 @ 58)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #2 0x7ffe148f6400 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #3 0x7ffe148f64c0 b   self-hosted:915 (0x7ffb086f12b8 @ 367)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #4 0x7ffe148f65b0 b   resource:///org/gnome/gjs/modules/signals.js:128 (0x7ffb086d3230 @ 386)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #5 0x55f9c6126c40 i   /usr/share/gnome-shell/extensions/ubuntu-appindicators@ubuntu.com/appIndicator.js:190 (0x7ffad4149120 @ 22)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #6 0x7ffe148f7200 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #7 0x55f9c6126b98 i   /usr/share/gnome-shell/extensions/ubuntu-appindicators@ubuntu.com/statusNotifierWatcher.js:219 (0x7ffad41456f8 @ 225)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #8 0x7ffe148f7de0 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #9 0x55f9c6126b20 i   /usr/share/gnome-shell/extensions/ubuntu-appindicators@ubuntu.com/extension.js:61 (0x7ffad413b9a0 @ 37)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #10 0x7ffe148f8ac0 b   resource:///org/gnome/shell/ui/extensionSystem.js:82 (0x7ffb0845a120 @ 431)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #11 0x55f9c6126aa0 i   resource:///org/gnome/shell/ui/extensionSystem.js:344 (0x7ffb0845abc0 @ 13)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #12 0x7ffe148f9830 b   self-hosted:251 (0x7ffb086c3ab0 @ 223)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #13 0x55f9c6126a20 i   resource:///org/gnome/shell/ui/extensionSystem.js:343 (0x7ffb0845ab38 @ 64)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #14 0x55f9c61269a0 i   resource:///org/gnome/shell/ui/extensionSystem.js:361 (0x7ffb0845ac48 @ 87)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #15 0x7ffe148fad30 b   resource:///org/gnome/gjs/modules/signals.js:128 (0x7ffb086d3230 @ 386)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #16 0x7ffe148fb540 b   resource:///org/gnome/shell/ui/sessionMode.js:205 (0x7ffb0806b670 @ 254)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #17 0x7ffe148fc930 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #18 0x55f9c6126860 i   resource:///org/gnome/shell/ui/sessionMode.js:167 (0x7ffb0806b450 @ 40)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #19 0x7ffe148fdc90 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #20 0x55f9c61267b8 i   resource:///org/gnome/shell/ui/screenShield.js:1280 (0x7ffb08050bc0 @ 188)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #21 0x7ffe148feff0 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #22 0x55f9c6126708 i   resource:///org/gnome/shell/ui/screenShield.js:1329 (0x7ffb08050c48 @ 391)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #23 0x7ffe14900350 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #24 0x55f9c6126680 i   resource:///org/gnome/shell/ui/shellDBus.js:462 (0x7ffb0805ad58 @ 67)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #25 0x7ffe149016b0 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #26 0x55f9c61265b8 i   resource:///org/gnome/gjs/modules/overrides/Gio.js:295 (0x7ffb086e5450 @ 614)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #27 0x55f9c6126510 i   resource:///org/gnome/gjs/modules/overrides/Gio.js:326 (0x7ffb086e5670 @ 34)
Jul  2 15:27:00 computer gnome-shell[2982]: Object .Gjs_AppIndicatorIconActor__1 (0x55f9c6604430), has been already finalized. Impossible to set any property to it.
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: == Stack trace for context 0x55f9c5c7a340 ==
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #0 0x7ffe148f58f0 I   resource:///org/gnome/gjs/modules/_legacy.js:83 (0x7ffb086b5de0 @ 87)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #1 0x55f9c6126cc8 i   /usr/share/gnome-shell/extensions/ubuntu-appindicators@ubuntu.com/indicatorStatusIcon.js:93 (0x7ffad4155450 @ 58)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #2 0x7ffe148f64d0 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #3 0x7ffe148f6590 b   self-hosted:915 (0x7ffb086f12b8 @ 367)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #4 0x7ffe148f6620 I   resource:///org/gnome/gjs/modules/signals.js:128 (0x7ffb086d3230 @ 386)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #5 0x55f9c6126c40 i   /usr/share/gnome-shell/extensions/ubuntu-appindicators@ubuntu.com/appIndicator.js:190 (0x7ffad4149120 @ 22)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #6 0x7ffe148f7200 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #7 0x55f9c6126b98 i   /usr/share/gnome-shell/extensions/ubuntu-appindicators@ubuntu.com/statusNotifierWatcher.js:219 (0x7ffad41456f8 @ 225)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #8 0x7ffe148f7de0 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #9 0x55f9c6126b20 i   /usr/share/gnome-shell/extensions/ubuntu-appindicators@ubuntu.com/extension.js:61 (0x7ffad413b9a0 @ 37)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #10 0x7ffe148f8ac0 b   resource:///org/gnome/shell/ui/extensionSystem.js:82 (0x7ffb0845a120 @ 431)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #11 0x55f9c6126aa0 i   resource:///org/gnome/shell/ui/extensionSystem.js:344 (0x7ffb0845abc0 @ 13)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #12 0x7ffe148f9830 b   self-hosted:251 (0x7ffb086c3ab0 @ 223)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #13 0x55f9c6126a20 i   resource:///org/gnome/shell/ui/extensionSystem.js:343 (0x7ffb0845ab38 @ 64)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #14 0x55f9c61269a0 i   resource:///org/gnome/shell/ui/extensionSystem.js:361 (0x7ffb0845ac48 @ 87)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #15 0x7ffe148fad30 b   resource:///org/gnome/gjs/modules/signals.js:128 (0x7ffb086d3230 @ 386)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #16 0x7ffe148fb540 b   resource:///org/gnome/shell/ui/sessionMode.js:205 (0x7ffb0806b670 @ 254)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #17 0x7ffe148fc930 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #18 0x55f9c6126860 i   resource:///org/gnome/shell/ui/sessionMode.js:167 (0x7ffb0806b450 @ 40)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #19 0x7ffe148fdc90 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #20 0x55f9c61267b8 i   resource:///org/gnome/shell/ui/screenShield.js:1280 (0x7ffb08050bc0 @ 188)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #21 0x7ffe148feff0 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #22 0x55f9c6126708 i   resource:///org/gnome/shell/ui/screenShield.js:1329 (0x7ffb08050c48 @ 391)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #23 0x7ffe14900350 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #24 0x55f9c6126680 i   resource:///org/gnome/shell/ui/shellDBus.js:462 (0x7ffb0805ad58 @ 67)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #25 0x7ffe149016b0 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7ffb086b5de0 @ 71)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #26 0x55f9c61265b8 i   resource:///org/gnome/gjs/modules/overrides/Gio.js:295 (0x7ffb086e5450 @ 614)
Jul  2 15:27:00 computer org.gnome.Shell.desktop[2982]: #27 0x55f9c6126510 i   resource:///org/gnome/gjs/modules/overrides/Gio.js:326 (0x7ffb086e5670 @ 34)
```

**Comment by LincolnR-IMR** [comment]
Me too (Ubuntu 18.04, extension installed from e.g.o)

The journal shows the file with the problem. In my case:
/home/USER/.local/share/gnome-shell/extensions/system-monitor@paradoxxx.zero.gmail.com/extension.js

To manually fix it:
* edit the file
* find the enable function
* add a line to re-create the Schema object
* save the file
* restart gnome-shell

The end result should look like this:

    var enable = function () {
        Schema = Convenience.getSettings();
        log('System monitor applet enabling');

**Comment by scr4bble** [comment]
WTF is this bug ?!
Please guys do something about it... it writes incredible amount of warnings into the log, makes it huge and even freezes a computer (for several seconds, sometimes minutes) when waking it up !!
It is not just that some graph is broken but it affects normal using of the computer.

It cannot be that hard to fix it and push live ...

**Comment by arigit** [comment]
any chance to get the updated code pushed to extensions.gnome.org?

**Comment by chrisspen** [comment]
I've been holding off a release for other 18.04 related bugs to be fixed, but if everyone says that master is working for them on 18.04, then I'll push a release. Note that EGO's turnaround time is about a month.

**Comment by arigit** [comment]
I'd suggest going ahead at least with this fix - this is creating journal sizes of several gigs and severely polluting the journal. The other issue I had to fix manually on the extension was the link to the "system monitor", in the context menu. First I had to adapt it to open the ubuntu-provided snap, and then I removed the snap and installed the regular .deb, and had to manually revert the change. Hoping that the lastest master can handle both configurations (snap and deb). The snap has theming issues.

**Comment by chrisspen** [comment]
I've uploaded release 36.

**Comment by bittner** [comment]
The extension update arrived today, thanks! Syslog is now silent.

> The other issue I had to fix manually on the extension was the link to the "system monitor", in the context menu.

Does that say, the link "System Monitor..." (above "Preferences...") should now be working? It still doesn't, I'm afraid. (Is there a separate issue for that?) - I'm on Ubuntu 18.04.1 LTS. This is what is printed to Syslog when the option is selected:

> Aug 24 09:26:19 bittner gnome-shell[2678]: JS ERROR: Exception in callback for signal: activate: TypeError: _gsmApp is null#012enable/<@/home/peter/.local/share/gnome-shell/extensions/system-monitor@paradoxxx.zero.gmail.com/extension.js:2436:13#012_emit@resource:///org/gnome/gjs/modules/signals.js:128:27#012activate@resource:///org/gnome/shell/ui/popupMenu.js:166:9#012wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22#012_onButtonReleaseEvent@resource:///org/gnome/shell/ui/popupMenu.js:127:9#012wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22

**Comment by bittner** [comment]
Anyway, I'm closing this issue as what I have reported is resolved.
